core/math/math_funcs.cpp

```diff
@@ -93,27 +93,18 @@ int Math::range_step_decimals(double p_step) {
 }
 
 double Math::ease(double p_x, double p_c) {
-	if (p_x < 0) {
-		p_x = 0;
-	} else if (p_x > 1.0) {
-		p_x = 1.0;
-	}
+	// Ensure that p_x stays within the range [0, 1]
+	p_x = (p_x < 0.0) ? 0.0 : ((p_x > 1.0) ? 1.0 : p_x);
```

Comment:
This is also doing redundant assignment, which is bad, it might be optimized out but it might not

Reply:
The `p_x = ` assignment existed before the PR, how would you suggest it done? I'm happy to change it

Reply:
> Or do you mean that before-hand it only assigned when out-of-range?

Yes, that's the unnecessary assignment that didn't occur before

Reply:
Ah, okay, I didn't really think of these little optimizations that were in place. Maybe your right and this PR has more cons thans pros. I can close if you'd like, unless you want a second opinion?

Reply:
It's up to you, at least this part should go as it hurts performance, but others might disagree